src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -2534,11 +2532,31 @@ void executeNodeActionsByKind(AnalysisScope analysisScope, ImmutableArray<Syntax
                     // and additionally the analyzer has not registered any code block start actions. In case
                     // the analyzer has registered code block start actions, we need to make callbacks for all nodes
                     // in the code block to ensure the analyzer can correctly report code block end diagnostics.
-                    var filteredNodesToAnalyze = declarationAnalysisData.IsPartialAnalysis && !groupedActionsForAnalyzer.HasCodeBlockStartActions
-                        ? nodesToAnalyze.WhereAsArray(analysisScope.ShouldAnalyze)
-                        : nodesToAnalyze;
+                    if (declarationAnalysisData.IsPartialAnalysis && !groupedActionsForAnalyzer.HasCodeBlockStartActions)
+                    {
+                        var filteredNodesToAnalyze = ArrayBuilder<SyntaxNode>.GetInstance(nodesToAnalyze.Count);
```

Comment:
No need to make a change, but I am curious if we should have a `WhereAsBuilder` API for creating populated ArrayBuilders from ImmutableArrays and ArrayBuilders.

Reply:
People may not like that it would wrap a call to GetInstance. But it seems like something we could get used to. Possibly with some further bikeshedding of the name.

Reply:
Yeah, i'm wary about making it too unclear about what's going on.  Could def see if these patterns become more prominant :)